src/main/Extensions/Base64Conversion/Base64Conversion.ts

```diff
@@ -0,0 +1,87 @@
+import { AssetPathResolver } from "@Core/AssetPathResolver";
+import { Extension } from "@Core/Extension";
+import { Translator } from "@Core/Translator";
+import { SearchResultItem, SearchResultItemActionUtility } from "@common/Core";
+import { Image } from "@common/Core/Image";
+import { Buffer } from "buffer";
+import { InvocationArgument } from "./InvocationArgument";
+
+export class Base64Conversion implements Extension {
+    public readonly id = "Base64Conversion";
+    public readonly name = "Base64 Conversion";
+
+    public readonly author = {
+        name: "Christopher Steiner",
+        githubUserName: "ChristopherSteiner",
+    };
+
+    private readonly defaultSettings = {};
+
+    public constructor(
+        private readonly assetPathResolver: AssetPathResolver,
+        private readonly translator: Translator,
+    ) {}
+
+    public async getSearchResultItems(): Promise<SearchResultItem[]> {
+        const { t } = this.translator.createT(this.getI18nResources());
+
+        return [
+            {
+                id: "Base64Converison:invoke",
+                description: t("searchResultItemDescription"),
+                name: t("searchResultItemName"),
+                image: this.getImage(),
+                defaultAction: SearchResultItemActionUtility.createInvokeExtensionAction({
+                    extensionId: this.id,
+                    description: t("searchResultItemActionDescription"),
+                    fluentIcon: "OpenRegular",
+                }),
+            },
+        ];
+    }
+
+    public async invoke(argument: InvocationArgument): Promise<string> {
+        if (argument.action === "encode") {
+            return Buffer.from(argument.payload, "binary").toString("base64");
+        } else {
+            return Buffer.from(argument.payload, "base64").toString("binary");
+        }
+    }
+
+    public isSupported(): boolean {
+        return true;
+    }
+
+    public getSettingDefaultValue<T>(key: string): T {
+        return this.defaultSettings[key];
```

Comment:
As your extension does not have any settings, and therefore also no default settings, you can just return undefined here:

```typescript
public getSettingDefaultValue() {
    return undefined;
}
```

And then you can also remove the private property `defaultSettings`.